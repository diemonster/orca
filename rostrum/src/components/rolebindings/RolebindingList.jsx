import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import RolebindingListItem from './RolebindingListItem';

export function RolebindingList(props) {
  const { rolebindings, selectedNamespace } = props;

  const rolebindingListItems = rolebindings.map(name => (
    <RolebindingListItem key={name} rolebindingName={name} />
  ));

  return (
    <div className="rolebinding-container">
      <h3>{`${selectedNamespace} rolebindings:`}</h3>
      <ul className="rolebinding-list">{rolebindingListItems}</ul>
    </div>
  );
}

RolebindingList.propTypes = {
  rolebindings: PropTypes.arrayOf(PropTypes.string),
  selectedNamespace: PropTypes.string.isRequired,
};

RolebindingList.defaultProps = {
  rolebindings: [],
};

const mapStateToProps = state => ({
  rolebindings: state.rolebinding.rolebindings,
  selectedNamespace: state.namespace.selectedNamespace,
});

export default connect(mapStateToProps)(RolebindingList);
