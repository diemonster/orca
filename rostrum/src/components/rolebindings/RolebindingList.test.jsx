import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { RolebindingList } from './RolebindingList';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('RolebindingList', () => {
    it('should render itself and its subcomponents when rolebindings.length === 0', () => {
      const props = {
        rolebindings: [],
        selectedNamespace: 'some-namespace',
      };

      const rolebindingList = shallow(<RolebindingList {...props} />);

      expect(rolebindingList).toMatchSnapshot();
    });

    it('should render itself and subcomponents when rolebindings.length > 0', () => {
      const props = {
        rolebindings: ['rolebinding1', 'rolebinding2'],
        selectedNamespace: 'some-namespace',
      };

      const rolebindingList = shallow(<RolebindingList {...props} />);

      expect(rolebindingList).toMatchSnapshot();
    });
  });
});
