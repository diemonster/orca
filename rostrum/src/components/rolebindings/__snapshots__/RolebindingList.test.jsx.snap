// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components RolebindingList should render itself and its subcomponents when rolebindings.length === 0 1`] = `
<div
  className="rolebinding-container"
>
  <h3>
    some-namespace rolebindings:
  </h3>
  <ul
    className="rolebinding-list"
  />
</div>
`;

exports[`components RolebindingList should render itself and subcomponents when rolebindings.length > 0 1`] = `
<div
  className="rolebinding-container"
>
  <h3>
    some-namespace rolebindings:
  </h3>
  <ul
    className="rolebinding-list"
  >
    <RolebindingListItem
      key="rolebinding1"
      rolebindingName="rolebinding1"
    />
    <RolebindingListItem
      key="rolebinding2"
      rolebindingName="rolebinding2"
    />
  </ul>
</div>
`;
